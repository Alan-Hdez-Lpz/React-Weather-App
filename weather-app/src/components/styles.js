import styled from 'styled-components';

export const Container = styled.div`
  padding: 2rem;
  display: flex;
  flex-direction: column;
  align-items: center;
`;

export const CardGrid = styled.div`
  display: flex;
  flex-wrap: wrap;
  gap: 1rem;
  justify-content: center;

  @media (max-width: 768px) {
    flex-direction: column;
    align-items: center;
  }
`;

export const WeatherCardStyled = styled.div`
  background-color: ${({ condition }) =>
    condition === 'Clear' ? '#ffeb3b' :
    condition === 'Rain' ? '#90caf9' :
    condition === 'Clouds' ? '#cfd8dc' :
    condition === 'Snow' ? '#e1f5fe' :
    '#e0e0e0'};
  color: #333;
  padding: 1.5rem;
  border-radius: 12px;
  width: 260px;
  box-shadow: 0 4px 10px rgba(0, 0, 0, 0.1);
  text-align: center;

  h2 {
    margin-bottom: 0.5rem;
  }

  @media (max-width: 768px) {
    width: 90%;
  }
`;
