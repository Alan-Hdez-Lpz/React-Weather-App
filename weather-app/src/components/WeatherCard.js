import { WeatherCardStyled } from './styles';

export default function WeatherCard({ weather }) {
  const condition = weather?.weather?.[0]?.main || 'Unknown';

  return (
    <WeatherCardStyled condition={condition}>
      <h2>{condition}</h2>
      <p>Temperature: {weather?.main?.temp ?? '--'} Â°C</p>
      <p>Humidity: {weather?.main?.humidity ?? '--'}%</p>
      <p>Wind: {weather?.wind?.speed ?? '--'} m/s</p>
    </WeatherCardStyled>
  );
}
