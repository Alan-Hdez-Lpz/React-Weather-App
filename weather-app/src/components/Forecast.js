import { useEffect, useState } from 'react';
import WeatherCard from './WeatherCard';
import { Container, CardGrid } from './styles';

const API_KEY = 'YOUR_API_KEY';

export default function Forecast() {
  const [cityName, setCity] = useState('Mexico city');
  const [query, setQuery] = useState('Mexico city');
  const [forecast, setForecast] = useState([]);
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState(null);

  useEffect(() => {
    async function fetchForecast(cityName) {
      setLoading(true);
    setError(null);
    try {
      const res = await fetch(
        `https://api.openweathermap.org/data/2.5/forecast?q=${cityName}&units=metric&appid=${API_KEY}`
      );
      if (!res.ok) throw new Error('City not found');
      const data = await res.json();
      const daily = data.list.filter(item => item.dt_txt.includes('12:00:00'));
      setForecast(daily);
    } catch (err) {
      setError(err.message);
      setForecast([]);
    } finally {
      setLoading(false);
    }
    }
    fetchForecast(cityName);
  }, [cityName]);

   const handleSearch = (e) => {
    e.preventDefault();
    if (query.trim()) {
      setCity(query.trim());
    }
  };

  return (
    <Container>
      <h1>5-Day Forecast</h1>

      <form onSubmit={handleSearch} style={{ marginBottom: '1rem' }}>
        <input
          type="text"
          placeholder="Enter city"
          value={query}
          onChange={(e) => setQuery(e.target.value)}
          style={{ padding: '0.5rem', fontSize: '1rem', marginRight: '0.5rem' }}
        />
        <button type="submit" style={{ padding: '0.5rem 1rem' }}>Search</button>
      </form>

      <h2>{cityName}</h2>

      {loading ? (
        <p>Loading forecast...</p>
      ) : error ? (
        <p style={{ color: 'red' }}>{error}</p>
      ) : (
        <CardGrid>
          {forecast.map(day => {
            const date = new Date(day.dt_txt);
            const weekday = date.toLocaleDateString('en-US', { weekday: 'long' });

            return (
              <div key={day.dt}>
                <h3>{weekday}</h3>
                <WeatherCard
                  weather={{
                    weather: day.weather,
                    main: day.main,
                    wind: day.wind,
                    name: cityName,
                  }}
                />
              </div>
            );
          })}
        </CardGrid>
      )}
    </Container>
  );
}
