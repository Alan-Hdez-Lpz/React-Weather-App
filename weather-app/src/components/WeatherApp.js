import { useState, useEffect } from 'react';
import WeatherCard from './WeatherCard';
import { Container } from './styles';

const API_KEY = 'YOUR_API_KEY';

export default function WeatherApp() {
  const [cityName, setCity] = useState('Mexico city');
  const [query, setQuery] = useState('Mexico city');
  const [weather, setWeather] = useState(null);
  const [loading, setLoading] = useState(true);
  const [error, setError] = useState(null);

  useEffect(() => {
    async function fetchWeather(cityName) {
        setLoading(true);
        setError(null);
        try {
        const res = await fetch(
            `https://api.openweathermap.org/data/2.5/weather?q=${cityName}&units=metric&appid=${API_KEY}`
        );
        if (!res.ok) throw new Error('City not found');
        const data = await res.json();
        setWeather(data);
        } catch (err) {
        setError(err.message);
        } finally {
        setLoading(false);
        }
    }
    fetchWeather(cityName);
  }, [cityName]);

  const handleSearch = (e) => {
    e.preventDefault();
    if (query.trim()) {
      setCity(query.trim());
    }
  };

  return (
    <Container>
      <h1>Current Weather</h1>
      <form onSubmit={handleSearch} style={{ marginBottom: '1rem' }}>
        <input
          type="text"
          placeholder="Enter city"
          value={query}
          onChange={(e) => setQuery(e.target.value)}
          style={{ padding: '0.5rem', fontSize: '1rem', marginRight: '0.5rem' }}
        />
        <button type="submit" style={{ padding: '0.5rem 1rem' }}>Search</button>
      </form>
      {loading ? (
        <p>Loading...</p>
      ) : error ? (
         <p style={{ color: 'red' }}>{error}</p>
      ) : (
        <>
          <h2>{weather.name}</h2>
          <WeatherCard weather={weather} />
        </>
      )}
    </Container>
  );
}
