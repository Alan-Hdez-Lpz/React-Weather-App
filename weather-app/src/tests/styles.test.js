import { render } from '@testing-library/react';
import { WeatherCardStyled } from '../components/styles';

test('WeatherCardStyled renders correctly', () => {
  const { container } = render(
    <WeatherCardStyled condition="Clear">Sunny</WeatherCardStyled>
  );
  expect(container.firstChild).toMatchSnapshot();
});
