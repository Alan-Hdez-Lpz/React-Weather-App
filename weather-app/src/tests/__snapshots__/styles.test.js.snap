// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherCardStyled renders correctly 1`] = `
<div
  class="sc-fhHczv kKUCfa"
  condition="Clear"
>
  Sunny
</div>
`;
